### ISSUE TITLE
java.lang.NullPointerException in DocumentsProvider (2)

### ISSUE THREAD

**Issue by jmue** [nextcloud/android#3973]
### Actual behaviour
- DocumentsProvider does not show folder content in some cases (app switching)

### Expected behaviour
- DocumentsProvider does not crash
 
### Steps to reproduce
1. Start Nextcloud app
2. Open Storagepicker/DocumentsUI
3. Navigate to a folder in third level hierarchy (e.g. '/Level1/Level2/')
4. Close Nextcloud with the app switcher
5. Try to move one level up in the Storagepicker via HeaderBar

fixes https://github.com/nextcloud/android/pull/3905

### Environment data
Android version: 8.1, 9.0

Device model: -

Stock or customized system:

Nextcloud app version: 3.6.1, master-branch

Nextcloud server version:

#### Web server error log
```
Insert your webserver log here
```

#### Nextcloud log (data/nextcloud.log)
```
2019-05-07 21:52:59.599 5888-5905/com.nextcloud.client E/DatabaseUtils: Writing exception to parcel
    java.lang.NullPointerException: Attempt to invoke virtual method 'android.accounts.Account com.owncloud.android.datamodel.FileDataStorageManager.getAccount()' on a null object reference
        at com.owncloud.android.providers.DocumentsStorageProvider.updateCurrentStorageManagerIfNeeded(DocumentsStorageProvider.java:574)
        at com.owncloud.android.providers.DocumentsStorageProvider.queryChildDocuments(DocumentsStorageProvider.java:161)
        at android.provider.DocumentsProvider.queryChildDocuments(DocumentsProvider.java:565)
        at android.provider.DocumentsProvider.query(DocumentsProvider.java:808)
        at android.content.ContentProvider$Transport.query(ContentProvider.java:240)
        at android.content.ContentProviderNative.onTransact(ContentProviderNative.java:102)
        at android.os.Binder.execTransact(Binder.java:731)
```
**NOTE:** Be super sure to remove sensitive data like passwords, note that everybody can look here! You can use the Issue Template application to prefill some of the required information: https://apps.nextcloud.com/apps/issuetemplate

**Comment by jmue** [comment]
![Screenshot_1557258242](https://user-images.githubusercontent.com/898577/57329795-b6b34f80-7114-11e9-90d8-0ff8568074a1.png)

**Comment by ezaquarii** [comment]
I confirmed this problem.

**Comment by ezaquarii** [comment]
```
    @SuppressLint("LongLogTag")
    private void updateCurrentStorageManagerIfNeeded(long docId) {
        if (rootIdToStorageManager == null) {
            try {
                queryRoots(FileCursor.DEFAULT_DOCUMENT_PROJECTION);
            } catch (FileNotFoundException e) {
                Log.e(TAG, "Failed to query roots");
            }
        }

        // This if fails to resolve docId to storage manager
        if (currentStorageManager == null ||
            rootIdToStorageManager.containsKey(docId) && currentStorageManager != rootIdToStorageManager.get(docId)) {
            currentStorageManager = rootIdToStorageManager.get(docId);
        }

        try {
            Account account = currentStorageManager.getAccount();
            OwnCloudAccount ocAccount = new OwnCloudAccount(account, MainApp.getAppContext());
            client = OwnCloudClientManagerFactory.getDefaultSingleton().getClientFor(ocAccount, getContext());
        } catch (OperationCanceledException | IOException | AuthenticatorException |
            com.owncloud.android.lib.common.accounts.AccountUtils.AccountNotFoundException e) {
            Log_OC.e(TAG, "Failed to set client", e);
        }
    }
```

See `// This if fails to resolve docId to storage manager` in the soruce snippet. I'm not sure how this is supposed to work, so maybe @tobiasKaminsky  or @mario can shed some light on a possible solution?

**Comment by ezaquarii** [comment]
I suspect that when we're going up, the `docId` of a subdirectory is simply not present in `rootIdStorageManager` map, that does not contain subdirs in it, and it fails.

**Comment by jmue** [comment]
This is already fixed with my PR #3905. But still not merged.

**Comment by ezaquarii** [comment]
@jmue still has CI build issues that needs to be fixed.

**Comment by ezaquarii** [comment]
@jmue It still has CI build issues that needs to be fixed. I'm keen on testing it using QA build. If you rebase, CI should publish QA apk for installation.

**Comment by jmue** [comment]
@ezaquarii I rebased my PR, so you can test it.  :thumbsup: I also answered to the review comments, but my comments get last after last rebase. So the point is where can I place them to keep them beyond the next rebase?

**Comment by ezaquarii** [comment]
@jmue Most reliable solution is to raise those comments again, or - better - place them as normal comments instead of review.

**Comment by AndyScherzinger** [comment]
> This is already fixed with my PR #3905. But still not merged.

@tobiasKaminsky can you give this one a review please? ❤️
